### PR TITLE
Attempting to get Angular working in JupyterLab.

### PR DESCRIPTION
**Pull request by danqrodney**

### CODE REVIEW

**Comment by danqrodney** [comment]
This just shows an empty concepts widget at this point. I still need to actually show the contents of the widget, make the API call, etc... which I may do before merging this PR. But I wanted to send this out for early feedback,

File: src/app/app.component.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+
+import {Component} from '@angular/core';
+
+// JupyterLab doesn't have custom webpack loaders. Need to be able to
+// inline the loaders so that they get picked up without having access to the
+// webpack.config.js file
+// See https://github.com/jupyterlab/jupyterlab/pull/4334#issuecomment-383104318
+import * as htmlTemplate from 'html-loader!./app.component.html';
```
Comment (calbach) [inline src/app/app.component.ts:33]: Oof.

File: src/app/app.component.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/*
+ *  Adapted from original source:
+ *  <https://github.com/SimonBiggs/scriptedforms/blob/master/scriptedforms/src/app/app.component.ts>
```
Comment (calbach) [inline src/app/app.component.ts:3]: The license on that repo is scary. I think this PR is OK, but wanted to flag that we should avoid any dependency on it.

**Review by calbach — COMMENTED** [review]
Pretty cool, but a lot of hacks in here. Based on what you've read, what's your sense of how supported or not supported this is (likelihood of breakage and how much sympathy we'd expect to receive from Angular/Jupyter folks)?

File: src/app/app.component.ts (hunk 2 above)
Comment (danqrodney) [reply src/app/app.component.ts:3]: Yeah, definitely. I've only been copying the code with the Apache license.

**Review by danqrodney — COMMENTED** [review]
TBH, I'm not sure how much support we will get regardless of approach. JupyterLab seems pretty new and raw at this point. There may be a higher chance that things will break with Angular, but I don't know enough at this point to say for sure. Perhaps JupyterLab will have official support for Angular at some point, though I haven't seen anything about it yet.

But this does worry me. If you think after looking at this it would be advisable to switch back and try to get non-Angular Typescript working again, please let me know.

**Review by calbach — APPROVED** [review]
I do worry about this breaking, but having Angular here is certainly nice. I could go either way, but I think this approach is good unless you hit unforeseen issues during implementation.

**Comment by danqrodney** [comment]
Abandoning Angular in favor of Typescript + React.

**Pull request closed without merging**